lib/versioning/gradle/compare.ts

```diff
@@ -192,6 +192,24 @@ export function compare(left: string, right: string): number {
   return 0;
 }
 
+export function valid(input: string): Token[] | null {
+  if (!input) {
+    return null;
+  }
+
+  if (!regEx(/^[-._+a-zA-Z0-9]+$/i).test(input)) {
+    return null;
+  }
+
+  if (regEx(/^latest\.?/i).test(input)) {
+    return null;
+  }
+
+  const tokens = tokenize(input);
+
+  return tokens;
```

Comment:
```suggestion
  return tokens?.length ? tokens : null;
```
Can `tokens` be empty? Then it's invalid.

Reply:
I haven't found a case where tokens will be empty.

Reply:
This change might create a coverage issue as there are no test cases where we get empty tokens.

Reply:
`isVersion` is using the same logic, how it's tested there?

Reply:
`  return !!tokens && !!tokens.length;`

This is the last statement in the `isVersion` function so the `token.length` method is executed but since there is no code after that we don't get coverage issues. I checked all the test cases none returned an empty array.

The issue seems to be very minor or non-existent though since the Codecov didn't throw an error.

Reply:
Ok, so we can expect to never get an empty array. Fine by me.

@rarkins WDYT?

Reply:
🤷

Reply:
@zharinov WDYT?

Reply:
I would like to use this check: 
```ts
// istanbul ignore if
if (!tokens?.length) {
  return null;
}
```

Reply:
Good idea, updated suggestion